backend/app/controllers/api/v1/courses_controller.rb

```diff
@@ -5,6 +5,11 @@ module V1
     class CoursesController < ApplicationController
       include Dry::Monads::Result::Mixin
 
+      def index
+        @courses = search_courses(Course.all)
+        render json: @courses
+      end
```

Comment:
1) Personally, I prefer to spit logic for searching records and preparing JSON.
2) Returns same result JSON for create and index actions

```ruby
def index
  @courses = search_courses(Course.all)
  return json: prepare_json_for_list(@courses)
end

def create
  case course_creation
  in Success(course)
    render json: prepare_json(course), status: :created
  # ...
end
private

def def search_courses(courses)
  if params[:search].present?
    courses.where('name LIKE ?', "%#{params[:search]}%") 
  else
    courses
  end
end

def prepare_json_for_list(courses)
  courses.map { |c|  prepare_json(c) }
end

def prepare_json(course)
  {
    id: course.id,
    name: course.name,
    user: {
      first_name: course.user.first_name
   }
  }
end
```

P.S. Maybe we need to return more data for course

Reply:
split